src/pages/groupComparison/GenericAssayEnrichmentCollections.tsx

```diff
@@ -0,0 +1,258 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import autobind from 'autobind-decorator';
+import { MolecularProfile } from 'cbioportal-ts-api-client';
+import { MakeMobxView } from '../../shared/components/MobxView';
+import Loader from '../../shared/components/loadingIndicator/LoadingIndicator';
+import ErrorMessage from '../../shared/components/ErrorMessage';
+import { MakeEnrichmentsTabUI } from './GroupComparisonUtils';
+import _ from 'lodash';
+import ComparisonStore from '../../shared/lib/comparison/ComparisonStore';
+import GenericAssayBinaryEnrichmentsContainer from 'pages/resultsView/enrichments/GenericAssayBinaryEnrichmentsContainer';
+import EnrichmentsDataSetDropdown from 'pages/resultsView/enrichments/EnrichmentsDataSetDropdown';
+import { observable } from 'mobx';
+import GenericAssayEnrichmentsContainer from 'pages/resultsView/enrichments/GenericAssayEnrichmentsContainer';
+import GenericAssayCategoricalEnrichmentsContainer from 'pages/resultsView/enrichments/GenericAssayCategoricalEnrichmentsContainer';
+
+export interface IGenericAssayEnrichmentCollectionsProps {
+    store: ComparisonStore;
+    genericAssayType: string;
+    resultsViewMode?: boolean;
+}
+
+@observer
+export default class GenericAssayEnrichmentCollections extends React.Component<
+    IGenericAssayEnrichmentCollectionsProps,
+    {}
+> {
+    @observable isBinary: boolean | undefined = false;
+    @observable isCategorical: boolean | undefined = false;
+    @observable isNumerical: boolean | undefined = false;
+
+    @autobind
+    private onChangeProfile(profileMap: {
+        [studyId: string]: MolecularProfile;
+    }) {
+        const type = Object.values(profileMap)[0].datatype;
+        this.props.store.setAllGenericAssayEnrichmentProfileMap(
+            profileMap,
+            this.props.genericAssayType
+        );
+        if (type == 'BINARY') {
```

Comment:
In JS/TS, we should use '===' instead of '==' because '==' doesn't check type equal, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality#description for more information

Reply:
I see, done!